### PR TITLE
Allow remark headings to be queried by level

### PR DESCRIPTION
**Pull request by thejameskyle**
Adds support for: `(depth: h1-6)`

```graphql
allMarkdownRemark {
  edges {
    node {
      headings(depth: h1) {
        value
      }
    }
  }
}
```

### CODE REVIEW

File: packages/gatsby-typegen-remark/package.json [hunk 1]
```diff
@@ -25,6 +27,7 @@
     "remark": "^6.2.0",
     "remark-html": "^5.0.1",
     "sanitize-html": "^1.13.0",
+    "underscore.string": "^3.3.4",
```
Comment (thejameskyle) [inline packages/gatsby-typegen-remark/package.json]: Added some missing dependencies

File: packages/gatsby-typegen-remark/src/gatsby-node.js [hunk 2]
```diff
@@ -180,6 +181,18 @@ exports.extendNodeType = ({ args, pluginOptions }) =>
...
 
+    const HeadingLevels = new GraphQLEnumType({
+      name: 'HeadingLevels',
+      values: {
+        'h1': { value: 1 },
+        'h2': { value: 2 },
+        'h3': { value: 3 },
+        'h4': { value: 4 },
+        'h5': { value: 5 },
+        'h6': { value: 6 },
```
Comment (thejameskyle) [inline packages/gatsby-typegen-remark/src/gatsby-node.js]: Had to use `h1-6` because graphql requires it.

**Comment by KyleAMathews** [comment]
Looks perfect, thanks!

**Comment by KyleAMathews** [comment]
Published this to NPM in 1.0.0-alpha12-alpha.1b1c742b 🎉

**Pull request merged**